src/pkg/cli/client/byoc/do/byoc.go

```diff
@@ -644,13 +644,32 @@ func (b *ByocDo) update(service composeTypes.ServiceConfig) *defangv1.ServiceInf
 	return si
 }
 
-func (b *ByocDo) setUp(ctx context.Context) error {
+func (b *ByocDo) needsSetup(ctx context.Context) (bool, error) {
 	projectCdImageTag, err := b.getCdImageTag(ctx)
+
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "no bucket found") {
```

Comment:
There's a helper in `s3.go` for this

Reply:
Actually, that's the fix: that function doesn't work for DO.

Reply:
I don't think I need to check this error anymore if I know I won't call this function before setup.